src/tools/run-make-support/src/rustdoc.rs

```diff
@@ -0,0 +1,59 @@
+use std::path::Path;
+use std::process::{Command, Output};
+
+use crate::{handle_failed_output, setup_common_build_cmd};
+
+pub fn rustdoc() -> RustdocInvocationBuilder {
+    RustdocInvocationBuilder::new()
+}
+
+#[derive(Debug)]
+pub struct RustdocInvocationBuilder {
+    cmd: Command,
+}
+
+impl RustdocInvocationBuilder {
+    fn new() -> Self {
+        let cmd = setup_common_build_cmd();
+        Self { cmd }
+    }
```

Comment:
Maybe missing a `-L $TARGET_RPATH_ENV`?

see https://github.com/rust-lang/rust/blob/4cdd20584ccb75890d7d9bfae266054abfae5d46/tests/run-make/tools.mk#L12

Reply:
Done. It *also* seems to set up `$(HOST_RPATH_ENV)` but I don't see that done in the rustc setup, so I *assume* it's not needed here?

Reply:
> Done. It _also_ seems to set up `$(HOST_RPATH_ENV)` but I don't see that done in the rustc setup, so I _assume_ it's not needed here?

I think it might be required if host/target differs, so it's safer if we make sure `HOST_RPATH_ENV` is available to `RUSTDOC` here.

```make
HOST_RPATH_ENV = \
    $(LD_LIB_PATH_ENVVAR)="$(TMPDIR):$(HOST_RPATH_DIR):$($(LD_LIB_PATH_ENVVAR))"
```